### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by SJTGSHIVAM**
Implemented user wishlist
- implemented cache invalidation on every wishlist related push
- please ignore catch prompts those will be added later
- live preview is available here: https://deploy-preview-4--log2-mock.netlify.app/

### CODE REVIEW

File: src/components/ProductCard.tsx [hunk 1]
```diff
@@ -79,7 +78,9 @@ export const ProductCard = ({
   );
   const addToCartThrottled = useCallback(throttle(addToCart, 1000), []);
 
-  useEffect(() => {}, [isInWishlist]);
+  useEffect(() => {
```
Comment (SJTGSHIVAM) [inline src/components/ProductCard.tsx:81]: This was missed in the last branch had a hard time figuring it out.

**Review by romabulani — APPROVED** [review]
- Clean and modular code
- Functionality is working as expected

Suggestion : If you want to add, you can add the number of items in wishlist on badge, its just a suggestion, its up to you

**Review by samyakshah3008 — APPROVED** [review]
Good easy logic for wishlist 
Clean code 

Good for approval ✅

**Comment by SJTGSHIVAM** [comment]
> 
@romabulani
Ahh! really missed it. I have the architecture ready it should be just a one-hour thing to gather all the pieces I guess. Will add it with the cart feature. Thanks for pointing.

**Review by thegeorgenikhil — COMMENTED** [review]
Suggestion: It would be better if the heart fills when you hover over it, as it would provide better feedback

Also you forgot to make the `cursor:pointer` on the heart icon

**Review by thegeorgenikhil — COMMENTED** [review]
Learnt about throttle function, really useful!

**Comment by SJTGSHIVAM** [comment]
> 

@thegeorgenikhil Thanks for mentioning I will look into it

**Review by thegeorgenikhil — APPROVED** [review]
Tried out the functionalities, works well!
Made some suggestions.
The code is modular.✅

**Comment by SJTGSHIVAM** [comment]
Hey, I am not sure but it shows a pointer on heart Icon for me. I will investigate a little deeper into it

**Pull request merged**